rustls-libssl/src/lib.rs

```diff
@@ -700,22 +807,21 @@ impl Ssl {
     }
 
     fn get_negotiated_cipher_suite_id(&self) -> Option<CipherSuite> {
-        self.conn
-            .as_ref()
+        self.conn()
             .and_then(|conn| conn.negotiated_cipher_suite())
             .map(|suite| suite.suite())
     }
 
     fn get_last_verification_result(&self) -> i64 {
-        if let Some(verifier) = &self.verifier {
-            verifier.last_result()
-        } else {
-            X509_V_ERR_UNSPECIFIED as i64
+        match &self.conn {
+            ConnState::Client(_, verifier) => verifier.last_result(),
+            ConnState::Server(_, verifier) => verifier.last_result(),
+            _ => X509_V_ERR_UNSPECIFIED as i64,
         }
     }
 
     fn get_error(&mut self) -> c_int {
-        match &mut self.conn {
+        match self.conn_mut() {
             Some(ref mut conn) => {
```

Comment:
pre-exsiting, but I think this `ref mut` is also unnecessary.

There's also one on L822 in `get_error`.